src/App.jsx

```diff
@@ -1,16 +1,66 @@
 import React from 'react';
-import { BrowserRouter, Route, Switch } from 'react-router-dom';
+import {
+  BrowserRouter,
+  Route,
+  Switch,
+  Redirect,
+} from 'react-router-dom';
+import Notifications from 'react-notify-toast';
 import Login from './components/login';
 import ReportPage from './components/ReportPage';
 
+const AuthenticatedRoute = ({ component: Component, authenticated, ...rest }) => (
+  <Route
+    {...rest}
+    render={
+      props => (
+        (authenticated === true) ? <Component {...props} {...rest} /> : <Redirect to="/login" />)
+    }
+  />
+);
+
 class App extends React.PureComponent {
+  state = {
+    authenticated: false,
+  }
+
+  setCurrentUser = (user) => {
+    if (user) {
```

Comment:
Instead of checking the `user` argument to determine what to set in the state, I suggest you make use of  `removeCurrentUser` if at any point in time you want to set `this.state.authenticated` to `false`.

Restricting this method to only setting the current user would help to ensure that the method is only doing what it is meant to and also help avoid duplicate codes.

Reply:
This is majorly for checking if a user has logged in in order to authenticate other pages. So I must check if a user exists.